### PR TITLE
[exporter/awskinesisexporter] Adding region config

### PR DESCRIPTION
**Pull request by MovieStoreGuy**
**Description:**
This was an oversight with the upgrade for the client to use `aws sdk v2`

**Link to tracking Issue:**
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16259

**Testing:**
NA

**Documentation:**
The documentation already exists.

### CODE REVIEW

**Review by bogdandrutu — APPROVED** [review]
This seem to be a bug, so please add a changelog entry

**Comment by MovieStoreGuy** [comment]
Apologies, I did forget it in my very tired state.

**Comment by simshengqin** [comment]
@Aneurysm9 any updates regarding this? I urgently require a fix to this issue and will appreciate an update, thank you!

**Review by Aneurysm9 — APPROVED** [review]

File: exporter/awskinesisexporter/exporter.go [hunk 1]
```diff
@@ -51,7 +51,16 @@ func createExporter(ctx context.Context, c component.ExporterConfig, log *zap.Lo
...
 	}
-	awsconf, err := awsconfig.LoadDefaultConfig(ctx)
+
+	var configOpts []func(*awsconfig.LoadOptions) error
+	if conf.AWS.Region != "" {
+		configOpts = append(configOpts, func(lo *awsconfig.LoadOptions) error {
+			lo.Region = conf.AWS.Region
+			return nil
+		})
+	}
```
Comment (codeboten) [inline exporter/awskinesisexporter/exporter.go:61]: How is this change tested?

Comment (MovieStoreGuy) [reply exporter/awskinesisexporter/exporter.go:61]: I've been thinking about this, I don't think I have a good answer atm. 

Let me play with some ideas on how to expose this information.

**Comment by MovieStoreGuy** [comment]
Stumbled with git here, sorry for the ping everyone.

File: exporter/awskinesisexporter/factory.go [hunk 2]
```diff
@@ -74,6 +74,7 @@ func NewTracesExporter(ctx context.Context, params component.ExporterCreateSetti
 		params,
 		conf,
 		exp.ConsumeTraces,
+		exporterhelper.WithStart(exp.Start),
```
Comment (MovieStoreGuy) [inline exporter/awskinesisexporter/factory.go:77]: I noticed that the start method wasn't being used, are you okay for me to include the fix in this PR @Aneurysm9 ?

Comment (Aneurysm9) [reply exporter/awskinesisexporter/factory.go:77]: Does that impact the ability to use the region config being added here?  It doesn't look like it to me, so I'd think to make this a separate change, but I could be missing something.

Comment (MovieStoreGuy) [reply exporter/awskinesisexporter/factory.go:77]: That makes sense with me, I just saw it and wanted to fix it up.

Let me move it to a seperate PR.

**Review by Aneurysm9 — APPROVED** [review]

**Comment by mx-psi** [comment]
@codeboten anything left to address here?

File: exporter/awskinesisexporter/exporter.go (hunk 1 above)
Comment (MovieStoreGuy) [reply exporter/awskinesisexporter/exporter.go:76]: So how I wanted to do this and what came out are very different due to the fact the fact that using `assert.EqualValues`  resulted in differences due to lambda references being stored.

The best alternative without doing much of a major refactor was to create an options type and redefine the `kinesis.NewClient` method.

**Comment by mx-psi** [comment]
ping @codeboten

**Review by codeboten — APPROVED** [review]
Looks good, thanks @MovieStoreGuy!

**Pull request merged**